.actions/assistant.py

```diff
@@ -49,86 +49,106 @@
 _PROJECT_ROOT = os.path.dirname(os.path.dirname(__file__))
 
 
-def _augment_requirement(ln: str, comment_char: str = "#", unfreeze: str = "all") -> str:
-    """Adjust the upper version contrains.
-
-    Args:
-        ln: raw line from requirement
-        comment_char: charter marking comment
-        unfreeze: Enum or "all"|"major"|""
-
-    Returns:
-        adjusted requirement
-
-    >>> _augment_requirement("arrow<=1.2.2,>=1.2.0  # anything", unfreeze="none")
-    'arrow<=1.2.2,>=1.2.0'
-    >>> _augment_requirement("arrow<=1.2.2,>=1.2.0  # strict", unfreeze="none")
-    'arrow<=1.2.2,>=1.2.0  # strict'
-    >>> _augment_requirement("arrow<=1.2.2,>=1.2.0  # my name", unfreeze="all")
-    'arrow>=1.2.0'
-    >>> _augment_requirement("arrow>=1.2.0, <=1.2.2  # strict", unfreeze="all")
-    'arrow>=1.2.0, <=1.2.2  # strict'
-    >>> _augment_requirement("arrow", unfreeze="all")
-    'arrow'
-    >>> _augment_requirement("arrow>=1.2.0, <=1.2.2  # cool", unfreeze="major")
-    'arrow>=1.2.0, <2.0  # strict'
-    >>> _augment_requirement("arrow>=1.2.0, <=1.2.2  # strict", unfreeze="major")
-    'arrow>=1.2.0, <=1.2.2  # strict'
-    >>> _augment_requirement("arrow>=1.2.0", unfreeze="major")
-    'arrow>=1.2.0, <2.0  # strict'
-    >>> _augment_requirement("arrow", unfreeze="major")
-    'arrow'
+class _RequirementWithComment(Requirement):
+    strict_string = "# strict"
+
+    def __init__(self, *args: Any, comment: str = "", pip_argument: Optional[str] = None, **kwargs: Any) -> None:
```

Comment:
This is not possible because a requirement might span multiple lines, for example

```
--extra-index-url https://download.pytorch.org/whl/cpu
torchdistx
```

However, the `_parse_requirements()` function below does what you propose and should be used for parsing instead of this class directly. This class is only meant to encapsulate the logic for composing the string considering whether it's strict and the comment